algosdk/atomic_transaction_composer.py

```diff
@@ -479,6 +479,55 @@ def submit(self, client: algod.AlgodClient) -> List[str]:
         self.status = AtomicTransactionComposerStatus.SUBMITTED
         return self.tx_ids
 
+    def simulate(
+        self, client: algod.AlgodClient
+    ) -> "SimulateAtomicTransactionResponse":
+
+        if self.status <= AtomicTransactionComposerStatus.SUBMITTED:
+            self.gather_signatures()
```

Comment:
I thought you could simulate without signing

Reply:
I think you can, but should I make this just build group? If they're using ATC they have to pass _some_ signer anyway

Reply:
I'm not sure. I'm trying to imagine what a dry-run like experience should look like with the new endpoint. We want to enable a path that involves minimal friction. Currently ATC's `add_transaction()` expects a `TransactionWithSigner`. Perhaps solving the problem is as simple as introducing a new `TransactionSigner` sub-class, something like `SimulateSkipSigner` which instead of actually signing leaves a marker that is later picked up by `gather_signatures` and the other downstream ATC methods which let the transaction through in the case of `/simulate`.

I'm leery about being too ambitious in this PR so I don't want to push too hard on this front for this review.

So probably it's best to build out the MVP for supporting simulate, but now is also a good time to create a design doc for making simulate as easy to use as dry run.

Reply:
A Dummy or PlaceholderSigner would probably make a good addition to all the SDKs, I've got one here for when I don't yet have any signers available for example https://github.com/algorand-devrel/hello-beaker/blob/master/src/App.tsx#L11-L17

Reply:
I'm in favor of a `EmptySigner` subclass now that submitting blank signature transactions is acceptable.

Reply:
https://github.com/algorand/py-algorand-sdk/issues/443 Filed an issue to track this one